tests/infra_test.go

```diff
@@ -173,12 +174,18 @@ var _ = Describe("Infrastructure", func() {
 
 		table.DescribeTable("should be rotated when deleted for ", func(secretName string) {
 			By("destroying the certificate")
-			secret, err := virtClient.CoreV1().Secrets(flags.KubeVirtInstallNamespace).Get(secretName, metav1.GetOptions{})
-			Expect(err).ToNot(HaveOccurred())
-			secret.Data = map[string][]byte{
-				"random": []byte("nonsense"),
-			}
-			_, err = virtClient.CoreV1().Secrets(flags.KubeVirtInstallNamespace).Update(secret)
+			err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
Can you please explain why there is a conflict updating the secret?

Reply:
I can't pin point exactly where it happens, but it's totally possible that k8s tried to update the object's annotations as part of it's own logic, and it collided with the test's update. 

what I mean is, it doesn't necessarily suggest a flaw in the test operation

Reply:
>Does it mean that all the updates we have in the tests should be surrounded with retry.RetryOnConflict?

It is currently commonly used on [some](https://github.com/k8snetworkplumbingwg/kubemacpool/search?q=retryonconflict&unscoped_q=retryonconflict) kubemacpool tests, but I wouldn't go as far as to changing all `update` apis. it should be used where needed and I could imagine a scenario where we would want to know if a conflict occurred.

>I also see we have in utils a method RetryIfModified.

Well [RetryOnConflict()](https://github.com/kubernetes/client-go/blob/master/util/retry/util.go#L103) and `RetryIfModified()` work very similarly, the main difference is that `RetryOnConflict` works with exponential retry mechanisms while `RetryIfModified` just tries 10 times..

> Why didn't you use it? Do you think its usages should be substituted with retry.RetryOnConflict?

I would personally prefer to change all to `RetryOnConflict`, but it won't be in this PR scope

Reply:
here's what the conflict occurs.

Our testsuite sets a very aggressive cert rotation interval during the tests in order to uncover any issues that might occur during rotation.

What happens is

- this test calls Get on the cert secret
- virt-operator operator writes to that secret
- this test attempts to write to the same secret.

The fact that we call Get on the secret then attempt to write the secret after virt-operator mutated the secret independently of the test causes the Update failure in the test.

Reply:
Thanks @davidvossel for making the scenario clearer.

considering this scenario, if we indeed performed `Get` in the middle of the cert rotation's checks, then re-`Get`ing is a good idea, i.e. the retryOnConflict is a good approach IMO